src/Agent.Sdk/SecretMasking/ISecretMasker.cs

```diff
@@ -0,0 +1,27 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+using System;
+using ValueEncoder = Microsoft.TeamFoundation.DistributedTask.Logging.ValueEncoder;
+using ISecretMaskerVSO = Microsoft.TeamFoundation.DistributedTask.Logging.ISecretMasker;
+
+namespace Agent.Sdk.SecretMasking;
+public interface ISecretMasker : ISecretMaskerVSO
+{
+    new void AddRegex(String pattern);
+
+    void AddCompiledRegex(String pattern);
```

Comment:
>AddCompiledRegex [](http://example.com/codeflow?start=9&length=16)

I'm not sure I see the need to add interface surface area just to provide for compiling .NET regexes. That's a pretty narrow detail (compiling regexes) to get this level of representation in the contract.

I am not aware of any significant risk of regression in specifying a compiled regex, by the way. Since there's no downside (that I'm aware of), I'd tend to just push for taking the optimization. having a control that allows the behavioral change to be introduced/tested carefully makes sense.

Reply:
@michaelcfanning, thank you. I refactored this a bit, added ability to pass any other options (in case if we will need it).